### PR TITLE
[Enhancement] make push down predicate scan rule more generic

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Right now `PushDownPredicateScanRule` only works on ES_SCAN_NODE / OLAP_SCAN_NODE, however this rule is much more generic. So I extend and simplify this method:
1. every `LogicalScanOperator` subclass has a `clone` method
2. In `PushDownPredicateScanRule`, we clone a scan operator and reset predicate. 

image

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java [hunk 1]
```diff
@@ -50,7 +50,12 @@ public LogicalScanOperator(
         this.table = Objects.requireNonNull(table, "table is null");
         this.colRefToColumnMetaMap = ImmutableMap.copyOf(colRefToColumnMetaMap);
         this.columnMetaToColRefMap = ImmutableMap.copyOf(columnMetaToColRefMap);
+        setPredicate(predicate);
+    }
 
+    @Override
+    public void setPredicate(ScalarOperator predicate) {
+        super.setPredicate(predicate);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java:58]: I think we must call `ColumnFilterConverter.convertColumnFilter` in specific rule, not always update it when set predicate or construncate a operator

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalMysqlScanOperator.java [hunk 2]
```diff
@@ -54,4 +54,9 @@ public LogicalMysqlScanOperator.Builder withOperator(LogicalMysqlScanOperator op
             return this;
         }
     }
+
+    @Override
+    public LogicalMysqlScanOperator clone() {
+        return new LogicalMysqlScanOperator.Builder().withOperator(this).build();
+    }
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalMysqlScanOperator.java:61]: Why you don't use OperatorBuilderFactory?

Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalMysqlScanOperator.java:61]: OK. will fix that.

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java|18|18|100.00%|[]|

**Review by DorianZheng — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by caneGuy — APPROVED** [review]

**Pull request merged**